### PR TITLE
Add a case study about `file`

### PR DESCRIPTION
**Pull request by jvoisin**
Signed-off-by: Julien Voisin <jvoisin@google.com>

### CODE REVIEW

File: doc/CaseStudies.md [hunk 1]
```diff
@@ -63,3 +63,24 @@ This way we were able to unblock the jsonnet fuzzer and increase the call tree c
...
+
+[MWDB](https://github.com/CERT-Polska/mwdb-core) uses [file](https://github.com/file/file)
+on malware samples, which is [worrying](https://github.com/CERT-Polska/mwdb-core/issues/671) to say the least.
+
+After sending [a]( https://github.com/google/oss-fuzz/pull/8536)
+[couple of](https://github.com/google/oss-fuzz/pull/8535)
+[pull requests](https://github.com/google/oss-fuzz/pull/8533) to tackle the low-hanging fruits,
+the [Remaining optimal interesting
+functions](https://storage.googleapis.com/oss-fuzz-introspector/file/inspector-report/20220901/fuzz_report.html#Analyses-and-suggestions)
+section showed that am awful lot of functions in the `readelf.c` file weren't touched
```
Comment (DavidKorczynski) [inline doc/CaseStudies.md:78]: 'am awful' --> 'an awful'

**Review by DavidKorczynski — APPROVED** [review]
Very neat!

File: doc/CaseStudies.md [hunk 2]
```diff
@@ -63,3 +63,24 @@ This way we were able to unblock the jsonnet fuzzer and increase the call tree c
...
+the [Remaining optimal interesting
+functions](https://storage.googleapis.com/oss-fuzz-introspector/file/inspector-report/20220901/fuzz_report.html#Analyses-and-suggestions)
+section showed that an awful lot of functions in the `readelf.c` file weren't touched
+at all by the fuzzers.
+
+Looking at the [Fuzz blockers](https://storage.googleapis.com/oss-fuzz-introspector/file/inspector-report/20220901/fuzz_report.html#fuzz_blocker0),
+`file_tryelf` was likely the functions that should be called, and by checking out the
+[coverage of the relevant file](https://storage.googleapis.com/oss-fuzz-coverage/file/reports/20220901/linux/src/file/src/funcs.c.html#L421),
+the culprit was that `file` needs to be passed data via a proper file descriptor
+to exercise its elf-related codepath, and thus a file-based fuzzer was [prompty added](https://github.com/google/oss-fuzz/pull/8542),
```
Comment (DavidKorczynski) [inline doc/CaseStudies.md:85]: prompty --> promptly

**Review by DavidKorczynski — CHANGES_REQUESTED** [review]

**Review by DavidKorczynski — APPROVED** [review]
lgtm -- thanks a lot for the case study!

**Comment by DavidKorczynski** [comment]
once it's ready for merge, please sign off the commits:

```
git rebase HEAD~3 --signoff
git push --force-with-lease origin add_file_casestudy
```

**Review by Navidem — APPROVED** [review]
Nicely done! Thanks!

**Pull request merged**